File: polyglot/csharp/SeisTotalCapabilityKernel.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Seis.Capability;

public enum SeisPlatform
{
    Windows,
    Android,
    Apple,
    Web,
    Data,
    Policy
}

public sealed record SeisCapability(
    string Name,
    SeisPlatform Platform,
    IReadOnlyCollection<string> LanguageFamilies,
    bool RequiresRuntimeInstall = false,
    bool MarketReady = false);

public static class SeisTotalCapabilityKernel
{
    private static readonly ISet<string> WindowsFirstLanguages = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
    {
        "C#", "F#", "PowerShell", "Batch", "C++", "Rust", "Go", "T-SQL", "MSBuild", "SQL"
    };

    private static readonly ISet<string> BlockedForThisPass = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
    {
        "JavaScript", "Python"
    };

    public static bool Accepts(SeisCapability capability)
    {
        if (capability.RequiresRuntimeInstall) return false;
        if (capability.LanguageFamilies.Any(BlockedForThisPass.Contains)) return false;
        if (capability.Platform != SeisPlatform.Windows) return true;
        return capability.LanguageFamilies.All(WindowsFirstLanguages.Contains);
    }

    public static string NextGate(SeisCapability capability)
    {
        if (!Accepts(capability))
        {
            return "Use Windows/non-Apple contracts or document product need, owner, validation, storage budget, and rollback.";
        }

        return capability.MarketReady
            ? "Ready for reviewed main-branch Windows release planning."
            : "Complete security, support, accessibility, observability, installation, and rollback readiness.";
    }
}

File: polyglot/kotlin/SeisTotalCapabilityKernel.kt
package seis.capability

enum class SeisPlatform { ANDROID, WINDOWS, APPLE, WEB, DATA, POLICY }

data class SeisCapability(
    val name: String,
    val platform: SeisPlatform,
    val languageFamilies: Set<String>,
    val requiresRuntimeInstall: Boolean = false,
    val marketReady: Boolean = false
)

object SeisTotalCapabilityKernel {
    private val androidFirstLanguages = setOf("Kotlin", "Java", "XML", "Gradle", "C++", "Rust", "Go", "SQL")
    private val blockedForThisPass = setOf("JavaScript", "Python")

    fun accepts(capability: SeisCapability): Boolean {
        if (capability.requiresRuntimeInstall) return false
        if (capability.languageFamilies.any { it in blockedForThisPass }) return false
        if (capability.platform != SeisPlatform.ANDROID) return true
        return capability.languageFamilies.all { it in androidFirstLanguages }
    }

    fun nextGate(capability: SeisCapability): String = when {
        !accepts(capability) -> "Use Android-native/non-Apple contracts or document runtime need, owner, checks, and rollback."
        !capability.marketReady -> "Finish emulator, accessibility, low-motion, privacy, security, support, and release readiness."
        else -> "Ready for reviewed main-branch Android release planning."
    }
}
